src/Symfony/Component/EventDispatcher/Debug/TraceableEventDispatcher.php

```diff
@@ -187,7 +187,7 @@ public function getCalledListeners(Request $request = null): array
     public function getNotCalledListeners(Request $request = null): array
     {
         try {
-            $allListeners = $this->getListeners();
+            $allListeners = (new \ReflectionProperty($this->dispatcher, 'listeners'))->getValue($this->dispatcher);
```

Comment:
we should go this path only if `$this->dispatcher instanceof EventDispatcher`, and the reflection should be created with `new \ReflectionProperty(EventDispatcher::class` to properly target the private var

Reply:
Done